app/uk/gov/hmrc/customs/declaration/connectors/declarationConnectors.scala

```diff
@@ -77,14 +76,14 @@ trait DeclarationConnector extends DeclarationCircuitBreaker with HttpErrorFunct
   override lazy val unstablePeriodDurationInMillis = config.declarationsCircuitBreakerConfig.unstablePeriodDurationInMillis
   override lazy val unavailablePeriodDurationInMillis = config.declarationsCircuitBreakerConfig.unavailablePeriodDurationInMillis
 
-  def send[A](xml: NodeSeq, date: Instant, correlationId: UUID, apiVersion: ApiVersion)(implicit vpr: ValidatedPayloadRequest[A]): Future[HttpResponse] = {
+  def send[A](xml: NodeSeq, date: Instant, correlationId: UUID, apiVersion: ApiVersion)(implicit vpr: ValidatedPayloadRequest[A], hc: HeaderCarrier): Future[HttpResponse] = {
     val config = Option(serviceConfigProvider.getConfig(s"${apiVersion.configPrefix}$configKey")).getOrElse(throw new IllegalArgumentException("config not found"))
     val bearerToken = "Bearer " + config.bearerToken.getOrElse(throw new IllegalStateException("no bearer token was found in config"))
-    implicit val hc: HeaderCarrier = HeaderCarrier(authorization = None)
+    implicit val headerCarrier: HeaderCarrier = HeaderCarrier(authorization = None)
```

Comment:
This line is bad and I know it wasn't written by you but I think this might be a good time to fix it. 

I'd be curious to know if authorisation = None actually does anything and if not then we can remove it as it is confusing.

One change that is easy and definitely needs to be made is to move this line to the post method below it. Currently we init the HeaderCarrier, do nothing with it, pass it down to another method then send it off. We should init it as the last possible moment which is below. 

Curious bit of code and this whole mess about headers needs cleaning up although that is outside of the scope of this ticket, but the issue we had is just a great example of why dodgy code like this leads to problems.